src/NumericalAlgorithms/SphericalHarmonics/YlmSpherepack.hpp

```diff
@@ -437,13 +448,14 @@ class YlmSpherepack {
                                 size_t physical_stride = 1,
                                 size_t physical_offset = 0,
                                 bool loop_over_offset = false) const;
-  void calculate_interpolation_data() const;
-  void fill_scalar_work_arrays() const;
-  void fill_vector_work_arrays() const;
+  void calculate_collocation_points();
+  void calculate_interpolation_data();
+  void fill_scalar_work_arrays();
+  void fill_vector_work_arrays();
   size_t l_max_, m_max_, n_theta_, n_phi_;
   size_t spectral_size_;
-  mutable YlmSpherepack_detail::MemoryPool memory_pool_;
-  mutable YlmSpherepack_detail::Storage storage_;
+  static thread_local YlmSpherepack_detail::MemoryPool memory_pool_;
```

Comment:
What happens when multiple YlmSpherepack objects are around? Won't they all use the same memory pool since it's static?

Can these scenarios be unit-tested?